setuptools/tests/test_config.py

```diff
@@ -581,3 +582,29 @@ def test_entry_points(self, tmpdir):
 
         with get_dist(tmpdir) as dist:
             assert dist.entry_points == expected
+
+saved_dist_init = _Distribution.__init__
+class TestExternalSetters:
+    # During creation of the setuptools Distribution() object, we call
+    # the init of the parent distutils Distribution object via
+    # _Distribution.__init__ ().
+    #
+    # It's possible distutils calls out to various keyword
+    # implementations (i.e. distutils.setup_keywords entry points)
+    # that may set a range of variables.
+    #
+    # This wraps distutil's Distribution.__init__ and simulates
+    # something during that call setting the
+    # "long_description_content_type", which we check is correctly
+    # copied by the trailing parts of the setuptools __init__ into the
+    # dist.metadata field.
+    def _fake_distribution_init(self, dist, attrs):
+        saved_dist_init(dist, attrs)
+        dist.long_description_content_type = 'text/something'
+        return None
+
+    @patch.object(_Distribution, '__init__', autospec=True)
```

Comment:
Is this actually how third party libraries extending `setuptools` are expected to set `metadata.long_description_type`? If so, I'm not sure that's something we should be supporting. `_Distribution` is not a public class, so doing anything to it involves relying on implementation details.

Reply:
This is simulating an external project that has set/accessed values via a distutils.setup_keywords entry-point in the unit test (which happens opaquely during _Distribution.__init__() call in dist.py).  I don't see it as exposing _Distribution details outside setuptools.